src/components/content/credential/Credential.tsx

```diff
@@ -0,0 +1,228 @@
+        csp: '' as CredentialVariables.csp,
+        description: '',
+        name: '',
+        timeToLive: 0,
+        type: '' as CredentialVariables.type,
+        variables: [],
+        xpanseUser: '',
+    });
+
+    const columns: ColumnsType<CredentialVariables> = [
+        {
+            title: 'Csp',
+            dataIndex: 'csp',
+        },
+        {
+            title: 'User',
+            dataIndex: 'xpanseUser',
+        },
+        {
+            title: 'Name',
+            dataIndex: 'name',
+        },
+        {
+            title: 'Type',
+            dataIndex: 'type',
+        },
+        {
+            title: 'Description',
+            dataIndex: 'description',
+        },
+        {
+            title: 'Operation',
+            dataIndex: 'operation',
+            render: (text: string, record: CredentialVariables) => {
+                return (
+                    <>
+                        <Space size='middle'>
+                            <Button type='primary' icon={<FullscreenOutlined />} onClick={() => details(record)}>
+                                Details
+                            </Button>
+                            <Popconfirm
+                                title='Delete the Credential'
+                                description='Are you sure to delete the Credential?'
+                                okText='Yes'
+                                cancelText='No'
+                                onConfirm={() => deleteCredential(record)}
+                            >
+                                <Button type='primary' icon={<MinusCircleOutlined />}>
+                                    delete
+                                </Button>
+                            </Popconfirm>
+                            <Button
+                                type='primary'
+                                icon={<InfoCircleOutlined />}
+                                onClick={() => updateCredential(record)}
+                            >
+                                update
+                            </Button>
+                        </Space>
+                    </>
+                );
+            },
+        },
+    ];
+
+    const addCredential = () => {
+        setIsAddOpen(true);
+    };
+
+    const deleteCredential = (record: CredentialVariables) => {
+        void CredentialsManagementService.deleteCredential(record.csp, record.xpanseUser, record.type)
+            .then(() => {
+                getTipInfo('success', 'delete success!');
+                getCredentials();
+            })
+            .catch(() => {
+                getTipInfo('error', 'delete failed!');
+            });
+    };
+
+    const updateCredential = (record: CredentialVariables) => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const createCredential: CreateCredential = JSON.parse(JSON.stringify(record));
```

Comment:
i would prefer to create the CreateCredential object using regular constructor instead of directly taking keys from CredentialVariables. We might in the future even change the fields in both of these types.

Reply:
@swaroopar Hi swaroopar ,Here you need to use deep copy for data object conversion, use to assign new object properties, when adding Credential, the newly added Credential cannot be queried

Reply:
we can do another GET to query the added cred
but in this case, i see we are building the request object. For this, i would avoid a deep copy

Reply:
i see the createCredentials is still not being explicitly still created.